verification-by-reference.md

```diff
@@ -0,0 +1,331 @@
+
+`registry.contoso.com/marketing/web:a2b2-verification`
+
+```json
+{
+  "schemaVersion": 2,
+  "manifests": [
+    {
+      "mediaType": "application/vnd.oci.image.config.v1+json",
+      "size": 7023,
+      "digest": "sha256:b5b2b2c507a0944348e0303114d8d93aaaa081732b864",
+      "signatures": [
+        {
+          "mediaType": "applicaiton/vnd.cncf.notary.x509-signature.config.v2",
+          "digest": "sha256:134c7fe821b9d359490cd009ce7ca322453f4f2d0",
+          "size": 1337,
+          "annotations": {
+            "org.opencontainers.image.signature.fingerprint": "43:51:43:a1:b5:fc:8b:b7:0a:3a:a9:b1:0f:66:73:a8"
+          }
+        }
+      ]
+    }
+  ]
+}
+```
+
+One challenge with this downward dependency approach is existing deployment documents which reference the original tag (`registry.contoso.com/marketing/web:a2b2`) and/or digest would need to change to the signed index reference (`registry.contoso.com/marketing/web:a2b2-verification`). This puts a new burden on a team that must change the references because another system may require a specific verification.
+
+Updating the existing tag could cause inconsistencies as some verification systems track the digest associated with a tag, and also limits new signatures from being pushed, without having to update the same index.
+
+## Reverse lookup
+
+**What if** we supported a reverse lookup, where a client can ask for all the verification objects for a given artifact? Reverse lookup refers to the ability to find which manifests are referenced by index objects.
+
+![](./media/oci-manifest-index-reverse-lookup.png)
+
+We could still use the oci index schema with two changes:
+
+1. Index is versioned to include `index.config` (artifactType in red)
+1. Add a new notary API that finds all indexes, with `"index.config.mediaType": "application/vnd.cncf.notary.verification.v1.config.json"`
+
+### web:a2b2 image as an OCI Manifest
+
+Same as above.
+
+```json
+{
+  "schemaVersion": 2,
+  "config": {
+    "mediaType": "application/vnd.oci.image.config.v1+json",
+    "size": 7023,
+    "digest": "sha256:b5b2b2c507a0944348e0303114d8d93aaaa081732b86451d9"
+  },
+  "layers": [
+    {
+      "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
+      "size": 32654,
+      "digest": "sha256:9834876dcfb05cb167a5c24953eba58c4ac89b1adf57f28"
+    },
+    {
+      "mediaType": "application/vnd.oci.image.layer.v1.tar+gzip",
+      "size": 16724,
+      "digest": "sha256:3c3a4604a545cdc127456d94e421cd355bca5b528f4a9c1"
+    }
+  ]
+}
+```
+
+### web:a2b2 staging verification w/ index
+
+An index is pushed that supports `index.config`, which tells the registry this index is a type of Notary Verification. This follows the same pattern we use with [OCI Artifacts to identify oci manifest as specific type](https://github.com/opencontainers/artifacts/blob/master/artifact-authors.md#defining-oci-artifact-types).
+This index would NOT be pushed with a tag. It would be a digest only reference. 
```

Comment:
Under Quay's current GC model, this would cause the new index to (eventually) be GCed, since it is unreferenced. We'd either have to keep around "reverse" references or whitelist these artifact types... I wonder if there is any upside to having these be tagged as well. If not, we'll just make the necessary changes on our end.